skxray/speckle_analysis.py

```diff
@@ -0,0 +1,279 @@
+
+    Returns
+    -------
+    roi_pix : dict
+        intensities of the ROI's of the labeled array according
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+
+    return {n: image[labels == n] for n in range(1, np.max(labels)+1)}
+
+
+def mean_intensity_sets(images_set, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for different image sets
+
+    Parameters
+    ----------
+    images : array
+        iterable of 4D arrays
+        shapes is: (len(images_sets), )
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_int_labels : dict
+        average intensity of each ROI as a dictionary
+        shape len(images_sets)
+        eg: 2 image sets,
+        {image set 1 : (len(images in image set 1), number of labels),
+        image set 2 : (len(images in image set 2), number of labels)}
+
+    """
+    return {n+1: mean_intensity(images_set[n],
+                                labels) for n in range(len(images_set))}
+
+
+def mean_intensity(images, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        iterable of 2D arrays
+        dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_intensity : array
+        mean intensity of each ROI for the set of images as an array
+        shape (len(images), number of labels)
+
+    """
+    if labels.shape != images[0].shape[0:]:
+        raise ValueError("Shape of the images should be equal to"
+                         " shape of the label array")
+
+    index = np.unique(labels)[1:]
+    mean_intensity = np.zeros((images.shape[0], index.shape[0]))
+
+    for n, img in enumerate(images):
+        mean_intensity[n] = mean(img, labels, index=index)
+
+    return mean_intensity
+
+
+def combine_mean_intensity(mean_int_dict):
+    """
+    Combine mean intensities of the images(all images sets) for each ROI
+
+    Parameters
+    ----------
+    mean_int_dict : dict
+        mean intensity of each ROI as a dictionary
+        eg: 2 image sets,
+        {image set 1 : (len(images in image set 1), number of labels),
+        image set 2 : (len(images in image set 2), number of labels)}
+
+    Returns
+    -------
+    combine_mean_int : array
+        combine mean intensities of image sets for each ROI of labeled array
+        shape (len(images in all image sets), number of labels)
+
+    """
+    return np.vstack(list(mean_int_dict.values()))
```

Comment:
This will come out in a random order in python3.  I assume this is to deal with the output from `mean_intensity_sets`, why not just have that return a list to begin with?